lib/models/profile.ts

```diff
@@ -15,13 +15,26 @@ import { Option } from "fp-ts/lib/Option";
 
 import { NonNegativeNumber } from "italia-ts-commons/lib/numbers";
 import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { BlockedChannel } from "../api/definitions/BlockedChannel";
 import { EmailAddress } from "../api/definitions/EmailAddress";
 import { FiscalCode } from "../api/definitions/FiscalCode";
 import { IsInboxEnabled } from "../api/definitions/IsInboxEnabled";
 import { IsWebhookEnabled } from "../api/definitions/IsWebhookEnabled";
 import { PreferredLanguages } from "../api/definitions/PreferredLanguages";
+import { ServiceId } from "../api/definitions/ServiceId";
 import { fiscalCodeToModelId } from "../utils/conversions";
 
+const ProfileBlockedChannels = t.dictionary(
+  ServiceId,
+  readonlySetType(BlockedChannel, "ProfileBlockedChannel"),
+  "ProfileBlockedChannels"
+);
+
+// typescript does not allow to have ServiceId as key type here
+export interface IProfileBlockedChannels {
+  readonly [serviceId: string]: ReadonlySet<BlockedChannel>;
```

Comment:
why not `ServiceId` instead of `string`?

Reply:
because typescript does not allow to have ServiceId as key type (only strings, as stated by the comment above : )